src/components/EditableField/EditableField.utils.ts

```diff
@@ -0,0 +1,78 @@
+import { FieldAction, FieldValue } from './EditableField.types'
+import { isArray, isObject } from '../../utilities/is'
+
+export const COMPONENT_KEY = 'EditableField'
+
+export const ACTION_ICONS = {
+  delete: 'cross-medium',
+  link: 'link',
+  plus: 'plus-medium',
+  valueOption: 'chevron-down',
+}
+
+export const deleteAction: FieldAction = {
+  name: 'delete',
+}
+
+export function normalizeFieldValue({
+  value,
+  name,
+  createNewFieldValue,
+  defaultOption,
+}): FieldValue[] {
+  let fieldValue: FieldValue[] = []
+
+  if (isArray(value)) {
+    fieldValue = value.map(val =>
+      createNewFieldValue({ value: val, name }, defaultOption)
+    )
+  } else {
+    fieldValue.push(createNewFieldValue({ value, name }, defaultOption))
+  }
+
+  return fieldValue
+}
+
+export function createNewValueFieldFactory(uuidFn) {
+  return function createNewValueFieldObject(
+    { value, name },
+    defaultOption?
+  ): FieldValue {
+    if (isObject(value)) {
+      const fieldObj = { ...value, id: uuidFn(`${name}_`) }
+
+      if (defaultOption != null && !Boolean(value.option)) {
```

Comment:
Should we give `defaultOption` a default value of `null` and then use a strict equality check for `!== null` here?

Reply:
yeah, I tend to use `something != null` because it checks for both `undefined` and `null`, strict comparison would just check `null`. In this case, I think making the default = `null` works, because I only care about `defaultOption` if it is defined. Thanks!